frontend/packages/console-shared/src/components/status/StatusIconAndText.tsx

```diff
@@ -0,0 +1,34 @@
+import * as React from 'react';
+import classNames from 'classnames';
+
+import { CamelCaseWrap } from '@console/internal/components/utils';
+import { DASH } from '../../constants';
+
+type StatusIconAndTextProps = {
+  icon?: React.ReactElement;
+  title?: string;
+  spin?: boolean;
+  iconOnly?: boolean;
+};
+
+const StatusIconAndText: React.FC<StatusIconAndTextProps> = ({ icon, title, spin, iconOnly }) => {
+  if (!title) {
```

Comment:
Just wondering, if `title` isn't defined but `icon` is, we'll still render a dash - is this an expected behavior?

Reply:
Probably not, I'll revisit.

Reply:
Yes, this is intentional as it allows to handle the case when `StatusIconAndText` component is used and the status (`title` prop) is not yet set. There is explicit `iconOnly` prop to render an icon without text.

Reply:
OK, thanks for clarification.